library/src/main/java/com/prolificinteractive/materialcalendarview/MaterialCalendarView.java

```diff
@@ -1306,6 +1326,41 @@ protected void dispatchOnDateSelected(final CalendarDay day, final boolean selec
     }
 
     /**
+     * Dispatch a range of days to a listener, if set
+     *
+     * @param firstDay first day enclosing a range
+     * @param lastDay  last day enclosing a range
+     */
+    protected void dispatchOnRangeSelected(final CalendarDay firstDay, final CalendarDay lastDay) {
+        final OnRangeSelectedListener listener = rangeListener;
+        final List<CalendarDay> days = new ArrayList<>();
+
+        final SortedSet<Date> sortedDays = new TreeSet<>();
```

Comment:
This sort is unnecessary--we only have two dates. If first day is after last day, call the method again with inverted params:

```
if (firstDay.isAfter(lastDay)) {
  dispatchOnRangeSelected(lastDay, firstDay);
  return;
}
```